nav2_bringup/launch/nav2_simulation_launch.py

```diff
@@ -16,207 +16,156 @@
+        'use_simulator',
+        default_value='True',
+        description='Whether to start the simulator')
+
+    declare_simulator_cmd = DeclareLaunchArgument(
         'simulator',
         default_value='gazebo',
         description='The simulator to use (gazebo or gzserver)')
 
-    declare_use_sim_time_cmd = launch.actions.DeclareLaunchArgument(
-        'use_sim_time',
-        default_value='true',
-        description='Use simulation (Gazebo) clock if true')
-
-    declare_use_simulation_cmd = launch.actions.DeclareLaunchArgument(
-        'use_simulation',
-        default_value='True',
-        description='Whether to run in simulation')
-
-    declare_world_cmd = launch.actions.DeclareLaunchArgument(
+    declare_world_cmd = DeclareLaunchArgument(
         'world',
-        default_value=os.path.join(get_package_share_directory('turtlebot3_gazebo'),
-                                   'worlds/turtlebot3_worlds/waffle.model'),
+        # TODO(orduno) Switch back once ROS argument passing has been fixed upstream
+        #              https://github.com/ROBOTIS-GIT/turtlebot3_simulations/issues/91
+        # default_value=os.path.join(get_package_share_directory('turtlebot3_gazebo'),
+        #                            'worlds/turtlebot3_worlds/waffle.model'),
+        default_value=os.path.join(bringup_dir, 'world/waffle.model'),
         description='Full path to world file to load')
 
-    stdout_linebuf_envvar = launch.actions.SetEnvironmentVariable(
-        'RCUTILS_CONSOLE_STDOUT_LINE_BUFFERED', '1')
-
     # Specify the actions
-    start_gazebo_cmd = launch.actions.ExecuteProcess(
-        condition=IfCondition(use_simulation),
+    start_gazebo_cmd = ExecuteProcess(
+        condition=IfCondition(use_simulator),
         cmd=[simulator, '-s', 'libgazebo_ros_init.so', world],
         cwd=[launch_dir], output='screen')
 
-    start_robot_state_publisher_cmd = launch.actions.ExecuteProcess(
-        cmd=[
-            os.path.join(
-                get_package_prefix('robot_state_publisher'),
-                'lib/robot_state_publisher/robot_state_publisher'),
-            os.path.join(
-                get_package_share_directory('turtlebot3_description'),
-                'urdf', 'turtlebot3_waffle.urdf'),
-            '--ros-args', '--params-file', configured_params],
-        cwd=[launch_dir], output='screen')
-
-    start_rviz_cmd = launch.actions.ExecuteProcess(
+    urdf = os.path.join(
+        get_package_share_directory('turtlebot3_description'), 'urdf', 'turtlebot3_waffle.urdf')
+
+    start_robot_state_publisher_cmd = Node(
+        package='robot_state_publisher',
+        node_executable='robot_state_publisher',
+        node_name='robot_state_publisher',
+        output='screen',
+        parameters=[{'use_sim_time': use_sim_time}],
+        arguments=[urdf])
+
+    # TODO(orduno) RVIZ crashing if launched as a node: https://github.com/ros2/rviz/issues/442
```

Comment:
Should this be updated to use the Node() syntax now? Looking at the issue linked, it appears the fix was merged, unless I'm misunderstanding the comments.

Reply:
I don't see any fix merged for the issue. I'm leaving the `ExecuteProcess` syntax.